jgrapht-core/src/main/java/org/jgrapht/traverse/CrossComponentIterator.java

```diff
@@ -55,12 +56,18 @@
     /**
      * Iterator which provides start vertices for cross-component iteration.
      */
+    private Iterator<V> entireGraphVertexIterator = null;
```

Comment:
What if the user calls setCrossComponent()? Clearly if it is changed to true we can just reinitialize the iterator to use the entire graph. But if they set it back to false, we would have no reference to what the originally supplied startVerticies we're.

The current set up allows for this even if may or may not make sense for a user to do this. And if the user shouldn't be able to do this perhaps we need to make it impossible.

I will do as you please, would you like me to break the functionality of being able to reset crossComponentTraversal to false?